pkg/operator/staticresourcecontroller/static_resource_controller.go

```diff
@@ -0,0 +1,212 @@
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), name),
+	}
+
+	c.operatorClient.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c *StaticResourceController) AddEventHandler(addEventHandler func(handler cache.ResourceEventHandler)) *StaticResourceController {
+	addEventHandler(c.eventHandler())
+	return c
+}
+
+func (c *StaticResourceController) AddNamespaceEventHandler(addEventHandler func(handler cache.ResourceEventHandler), namespaces ...string) *StaticResourceController {
+	interestingNamespaces := sets.NewString(namespaces...)
+	addEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				ns, ok := obj.(*corev1.Namespace)
+				if !ok {
+					c.queue.Add(workQueueKey)
+				}
+				if interestingNamespaces.Has(ns.Name) {
+					c.queue.Add(workQueueKey)
+				}
+			},
+			UpdateFunc: func(old, new interface{}) {
+				ns, ok := old.(*corev1.Namespace)
+				if !ok {
+					c.queue.Add(workQueueKey)
+				}
+				if interestingNamespaces.Has(ns.Name) {
+					c.queue.Add(workQueueKey)
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				ns, ok := obj.(*corev1.Namespace)
+				if !ok {
+					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+					if !ok {
+						utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %#v", obj))
+						return
+					}
+					ns, ok = tombstone.Obj.(*corev1.Namespace)
+					if !ok {
+						utilruntime.HandleError(fmt.Errorf("tombstone contained object that is not a Namespace %#v", obj))
+						return
+					}
+				}
+				if interestingNamespaces.Has(ns.Name) {
+					c.queue.Add(workQueueKey)
+				}
+			},
+		})
+	return c
+}
+
+func (c StaticResourceController) sync() error {
+	errors := []error{}
+	directResourceResults := resourceapply.ApplyDirectly(c.clients, c.eventRecorder, c.manifests, c.files...)
+	for _, currResult := range directResourceResults {
+		if currResult.Error != nil {
+			errors = append(errors, fmt.Errorf("%q (%T): %v", currResult.File, currResult.Type, currResult.Error))
+			continue
+		}
+	}
+
+	if len(errors) > 0 {
+		message := ""
+		for _, err := range errors {
+			message = message + err.Error() + "\n"
+		}
+		errors = append(errors,
+			appendErrors(v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Type:    fmt.Sprintf("%sDegraded", c.name),
+				Status:  operatorv1.ConditionTrue,
+				Reason:  "SyncError",
+				Message: message,
+			})))...,
+		)
+	} else {
+		errors = append(errors,
+			appendErrors(v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+				Type:   fmt.Sprintf("%sDegraded", c.name),
+				Status: operatorv1.ConditionFalse,
+				Reason: "AsExpected",
+			})))...,
+		)
+	}
+
+	return utilerrors.NewAggregate(errors)
+}
+
+func appendErrors(_ *operatorv1.OperatorStatus, _ bool, err error) []error {
```

Comment:
this method doesn't append errors, it converts an `err` to `[]error` :P

Reply:
> this method doesn't append errors, it converts an `err` to `[]error` :P

sure, I'll rename.